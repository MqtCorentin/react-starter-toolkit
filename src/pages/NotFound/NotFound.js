import ResiliencePage from 'shared/components/ResiliencePage';
import { TITLE, MESSAGE, CODE } from './constants';

const NotFound = () => <ResiliencePage title={TITLE} message={MESSAGE} code={CODE} subtitlePartOne="not" subtitlePartTwo="found" />;

export default NotFound;
