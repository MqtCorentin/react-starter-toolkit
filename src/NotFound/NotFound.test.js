import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import NotFound from './NotFound.component';

const wrapper = ({ children }) => <StaticRouter context={{}}>{children}</StaticRouter>;

describe('<NotFound page/>', () => {
  it('Renders NotFound page component without crashing', () => {
    const { asFragment } = render(<NotFound />, { wrapper });
    expect(asFragment()).toMatchSnapshot();
  });
});
