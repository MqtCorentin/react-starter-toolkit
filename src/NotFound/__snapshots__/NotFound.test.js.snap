// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound page/> Renders NotFound page component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--hasstepper af-title-bar--backhome"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        class="btn af-btn--circle"
        href="/"
        title="Retour Ã  l'accueil"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      404 Page Not Found
    </div>
  </div>
  <div
    class="container af-container--notfound"
  >
    <h1
      class="af-notfound__title"
    >
      <div
        class="af-notfound__title-covernumber"
      >
        <span
          class="af-notfound__title-number"
        >
          404
        </span>
      </div>
      <div
        class="af-notfound__title-covernot"
      >
        <span
          class="af-notfound__title-not"
        >
          not
          <br />
          found
        </span>
      </div>
    </h1>
    <p
      class="af-notfound__message"
    >
      The page you are looking for is not here!
    </p>
  </div>
</DocumentFragment>
`;
